server/util/validation_decorators/validate_project.py

```diff
@@ -0,0 +1,63 @@
+from db_models.pluser import plUser
+from db_models.project import Project
+from db_models.industries import Industry
+from flask import jsonify, request
+from functools import wraps
+from app import db
+from sqlalchemy import or_
+import re
+
+
+def validate_project(f):
+    @wraps(f)
+    def wrapper(*args, **kwargs):
+
+        title = request.json.get('title', None)
+        subtitle = request.json.get('subtitle', None)
+        location = request.json.get('location', None)
+        photos = request.json.get('photos', None)
+        industry = request.json.get('industry', None)
+        goal = request.json.get('funding_goal', None)
+        deadline = request.json.get('deadline', None)
+
+        if not title or not subtitle or not location or not goal or not deadline:
+            return jsonify({'error': 'Invalid JSON request'}), 400
+
+        # Verify Length of Title
+        if len(location) > 64:
+            return jsonify({'error': 'Location must be between less than 64 characters'}), 400
+
+        # Verify photos is in proper form, an array of strings
+        if type(photos) is not list:
+            return jsonify({'error': "Photos must be an array or list"}), 400
+
+        for photo in photos:
+            if type(photo) is not str:
+                return jsonify({'error': 'One of the photos is not a string'}), 400
+
+        # Verify industry is in proper form, an array of strings
+        if type(industry) is not list:
+            return jsonify({'error': "Industry must be an array or list"}), 400
+
+        for indus in industry:
+            if type(indus) is not str:
+                return jsonify({'error': 'One of the industries is not a string'}), 400
+
+        # Verify Industry exists
+        # Industry should be an array of string
+        filters = [Industry.name == i for i in industry]
+        if len(db.session.query(Industry).filter(or_(*filters)).all()) != len(industry):
+            return jsonify({"error": "That industry does not exist, please contact us to add an additional industry or choose one from the list"}), 400
+
+        # Verify goal is a float
+        try:
+            float(goal)
+        finally:
```

Comment:
I think you want "except" instead of "finally" here as code in a finally block will always execute

Reply:
Ah, thank you for the catch